addons/mail/models/mail_role.py

```diff
@@ -0,0 +1,17 @@
+from odoo.exceptions import ValidationError
+from odoo import _, api,fields, models
+
+
+class MailRole(models.Model):
+    _name = 'mail.role'
+    _description = "Role in Mail"
+
+    name = fields.Char(required=True, translate=True)
+    user_ids = fields.Many2many("res.users", string="Users")
+
+
+    @api.constrains('name')
+    def _check_name(self):
+        for record in self:
+            if record.search_count([('name', '=', record.name)]) > 1:
+                raise ValidationError(_("A role with the same name already exists."))
```

Comment:
This could be a sql index "unique":

```suggestion
   _unique_source_url = models.UniqueIndex("(name)", _("A role with the same name already exists."))
```